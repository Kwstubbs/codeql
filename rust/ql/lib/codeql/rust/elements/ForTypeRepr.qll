// generated by codegen, do not edit
/**
 * This module provides the public class `ForTypeRepr`.
 */

private import internal.ForTypeReprImpl
import codeql.rust.elements.ForBinder
import codeql.rust.elements.TypeRepr

/**
 * A type with a higher-ranked `for` modifier. This is currently not valid Rust syntax (`for<...>` can
 * only be applied to traits to form a `TypeBound`).
 *
 * For example:
 * ```rust
 * fn foo(value: for<'a> usize) {}  // DOESN'T COMPILE
 * //            ^^^^^^^^^^^^^
 * ```
 */
final class ForTypeRepr = Impl::ForTypeRepr;
