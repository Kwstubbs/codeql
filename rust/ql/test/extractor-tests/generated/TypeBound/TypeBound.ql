// generated by codegen, do not edit
import codeql.rust.elements
import TestUtils

query predicate instances(
  TypeBound x, string isAsync__label, string isAsync, string isConst__label, string isConst,
  string hasLifetime__label, string hasLifetime, string hasTypeRepr__label, string hasTypeRepr,
  string hasUseBoundGenericArgs__label, string hasUseBoundGenericArgs
) {
  toBeTested(x) and
  not x.isUnknown() and
  isAsync__label = "isAsync:" and
  (if x.isAsync() then isAsync = "yes" else isAsync = "no") and
  isConst__label = "isConst:" and
  (if x.isConst() then isConst = "yes" else isConst = "no") and
  hasLifetime__label = "hasLifetime:" and
  (if x.hasLifetime() then hasLifetime = "yes" else hasLifetime = "no") and
  hasTypeRepr__label = "hasTypeRepr:" and
  (if x.hasTypeRepr() then hasTypeRepr = "yes" else hasTypeRepr = "no") and
  hasUseBoundGenericArgs__label = "hasUseBoundGenericArgs:" and
  if x.hasUseBoundGenericArgs()
  then hasUseBoundGenericArgs = "yes"
  else hasUseBoundGenericArgs = "no"
}

query predicate getLifetime(TypeBound x, Lifetime getLifetime) {
  toBeTested(x) and not x.isUnknown() and getLifetime = x.getLifetime()
}

query predicate getTypeRepr(TypeBound x, TypeRepr getTypeRepr) {
  toBeTested(x) and not x.isUnknown() and getTypeRepr = x.getTypeRepr()
}

query predicate getUseBoundGenericArgs(TypeBound x, UseBoundGenericArgs getUseBoundGenericArgs) {
  toBeTested(x) and not x.isUnknown() and getUseBoundGenericArgs = x.getUseBoundGenericArgs()
}
