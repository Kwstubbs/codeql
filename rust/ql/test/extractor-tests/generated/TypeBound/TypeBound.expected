instances
| gen_type_bound.rs:7:15:7:19 | TypeBound | isAsync: | no | isConst: | no |
getForBinder
getLifetime
getTypeRepr
| gen_type_bound.rs:7:15:7:19 | TypeBound | gen_type_bound.rs:7:15:7:19 | Debug |
getUseBoundGenericArgs
